common/test/acceptance/tests/studio/test_studio_html_editor.py

```diff
@@ -100,3 +113,186 @@ def test_link_plugin_sets_url_correctly(self):
             static_link,
             "URL in the link plugin is different"
         )
+
+    def test_tinymce_and_codemirror_preserve_style_tags(self):
+        """
+        Scenario: TinyMCE and CodeMirror preserve style tags
+        Given I have created a Blank HTML Page
+            When I edit the page
+            And type "<p class='title'>pages</p><style><!-- .title { color: red; } --></style>" in the code editor and
+            press OK
+            And I save the page
+                Then the page text contains:
+                  ""
+                  <p class="title">pages</p>
+                  <style><!--
+                  .title { color: red; }
+                  --></style>
+                  ""
+        """
+        content = '<p class="title">pages</p><style><!-- .title { color: red; } --></style>'
+        self._content_test(content)
```

Comment:
Ideally the assertions should be in tests instead of in helper functions os that a user can understand the hole purpose of test just by looking at it's code